### PR TITLE
feat: show customer as logged-in in embedded checkout form

### PR DESCRIPTION
**Pull request by dhargitai**
…if they logged in inside VSF

## Description
- Show better thank you and error messages on checkout page
- Logout locally and redirect to home page if the user logs out inside the embedded checkout form
- Load embedded checkout URL from the cart composable instead of a cookie
- Set cookies that come from the Login API in the browser so the embedded checkout form should show the user as logged in

## Related Issue
VBC-15

## Motivation and Context
This is an enhancement of the functionality of VBC-15.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by scernisevs — APPROVED** [review]
LGTM :)

File: packages/api-client/src/helpers/cartResponse/index.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    | GetCartResponse
+    | UpdateLineItemResponse
+    | AddLineItemsResponse
+): void {
+  const {
+    config: {
+      sdkSettings: { guestToken }
+    }
+  } = context;
+  let embeddedCheckoutUrl = (response as any)?.data?.redirect_urls
```
Comment (WojtekTheWebDev) [inline packages/api-client/src/helpers/cartResponse/index.ts:23]: Is there any chance to type the `response`?

File: packages/composables/src/useUser/params.ts [hunk 2]
```diff
@@ -49,15 +48,10 @@ const params: UseUserFactoryParams<User, UpdateParams, RegisterParams> = {
...
-    context.$bigcommerce.config.app.$cookies.remove(COOKIE_KEY_CUSTOMER_DATA);
-
-    const newCart = await loadCart(context, {});
-    context.cart.setCart(newCart);
-  },
+  logOut,
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  updateUser: async (context: Context, { currentUser, updatedUserData }): Promise<User> => {
+  updateUser: async (context: Context): Promise<User> => {
```
Comment (WojtekTheWebDev) [inline packages/composables/src/useUser/params.ts:54]: Why this change?

File: packages/composables/src/useUser/params.ts [hunk 3]
```diff
@@ -67,7 +61,7 @@ const params: UseUserFactoryParams<User, UpdateParams, RegisterParams> = {
   logIn,
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  changePassword: async (context: Context, { currentUser, currentPassword, newPassword }): Promise<User> => {
+  changePassword: async (context: Context): Promise<User> => {
```
Comment (WojtekTheWebDev) [inline packages/composables/src/useUser/params.ts:64]: Why this change?

**Review by WojtekTheWebDev — CHANGES_REQUESTED** [review]
Three small questions from me

File: packages/composables/src/useUser/params.ts (hunk 2 above)
Comment (dhargitai) [reply packages/composables/src/useUser/params.ts:54]: These methods will be implemented in another task, the formatter is wrapping them into multiple lines because of the line length and then the ESLint complains about the unused variables.

I could place `// eslint-disable-next-line` on the wrapped lines but this seemed to be easier.

Comment (WojtekTheWebDev) [reply packages/composables/src/useUser/params.ts:54]: K thanks

**Review by WojtekTheWebDev — APPROVED** [review]
LGTM
